auth/ante_test.go

```diff
@@ -0,0 +1,367 @@
+	require.NoError(t, acc2.SetAccountNumber(100))
+	happ.AccountKeeper.SetAccount(ctx, acc2)
+
+	// msg and signatures
+	var tx sdk.Tx
+	msg1 := sdkAuth.NewTestMsg(addr1)
+	msg2 := sdkAuth.NewTestMsg(addr2)
+
+	acc1 = happ.AccountKeeper.GetAccount(ctx, hmTypes.AccAddressToHeimdallAddress(addr1))
+	acc2 = happ.AccountKeeper.GetAccount(ctx, hmTypes.AccAddressToHeimdallAddress(addr2))
+	// accNumber1 := acc1.GetAccountNumber()
+	// accNumber2 := acc2.GetAccountNumber()
+
+	// test good tx from one signer
+	tx = types.NewTestTx(ctx, msg1, priv1, uint64(0), uint64(0))
+	checkValidTx(t, anteHandler, ctx, tx, false)
+
+	// test good tx from one signer
+	tx = types.NewTestTx(ctx, msg1, priv1, uint64(0), uint64(1))
+	checkValidTx(t, anteHandler, ctx, tx, false)
+
+	// // new tx from wrong account number
+	tx = types.NewTestTx(ctx, msg2, priv2, uint64(1), uint64(1))
+	checkInvalidTx(t, anteHandler, ctx, tx, false, sdk.CodeUnauthorized)
+
+	// from correct account number but wrong private key
+	tx = types.NewTestTx(ctx, msg2, priv1, uint64(1), uint64(0)) // with private key 1
+	checkInvalidTx(t, anteHandler, ctx, tx, false, sdk.CodeUnauthorized)
+
+	// from correct account number but wrong private key
+	tx = types.NewTestTx(ctx, msg2, priv2, uint64(0), uint64(0)) // with private key 2 (account 2)
+	checkValidTx(t, anteHandler, ctx, tx, false)
+}
+
+func (suite *AnteTestSuite) TestSequences() {
+	t, happ, ctx, anteHandler := suite.T(), suite.app, suite.ctx, suite.anteHandler
+	ctx = ctx.WithBlockHeight(1)
+
+	// keys and addresses
+	priv1, _, addr1 := sdkAuth.KeyTestPubAddr()
+	priv2, _, addr2 := sdkAuth.KeyTestPubAddr()
+
+	// set the accounts, we don't need the acc numbers as it is in the genesis block
+	acc1 := happ.AccountKeeper.NewAccountWithAddress(ctx, hmTypes.AccAddressToHeimdallAddress(addr1))
+	acc1.SetCoins(simulation.RandomFeeCoins())
+	happ.AccountKeeper.SetAccount(ctx, acc1)
+	acc2 := happ.AccountKeeper.NewAccountWithAddress(ctx, hmTypes.AccAddressToHeimdallAddress(addr2))
+	acc2.SetCoins(simulation.RandomFeeCoins())
+	require.NoError(t, acc2.SetAccountNumber(100))
+	happ.AccountKeeper.SetAccount(ctx, acc2)
+
+	// msg and signatures
+	var tx sdk.Tx
+	msg1 := sdkAuth.NewTestMsg(addr1)
+	msg2 := sdkAuth.NewTestMsg(addr2)
+
+	acc1 = happ.AccountKeeper.GetAccount(ctx, hmTypes.AccAddressToHeimdallAddress(addr1))
+	acc2 = happ.AccountKeeper.GetAccount(ctx, hmTypes.AccAddressToHeimdallAddress(addr2))
+	accNumber1 := acc1.GetAccountNumber()
+	accNumber2 := acc2.GetAccountNumber()
+
+	// test good tx from one signer
+	tx = types.NewTestTx(ctx, msg1, priv1, accNumber1, uint64(0))
+	checkValidTx(t, anteHandler, ctx, tx, false)
+
+	// test good tx from one signer
+	tx = types.NewTestTx(ctx, msg1, priv1, accNumber1, uint64(1))
+	checkValidTx(t, anteHandler, ctx, tx, false)
+
+	// // new tx from wrong account number
+	tx = types.NewTestTx(ctx, msg2, priv2, accNumber2, uint64(1))
+	checkInvalidTx(t, anteHandler, ctx, tx, false, sdk.CodeUnauthorized)
```

Comment:
Dont we want to check for a specific error?

Reply:
`checkInvalidTx` is doing the same. It will validate `sdk. CodeUnauthorized` error.